HDF5 header files

I would like to build the NetCDF library which requires HDF5. Unfortunately, the current HDF5_jll, does not install the header files. The windows msys2 package (http://repo.msys2.org/mingw/i686/mingw-w64-i686-hdf5-1.10.5-1-any.pkg.tar.xz) does actually include the necessary header files, but the python whl do not. But apparently, they correspond to HDF5 1.10.4, and presumably they can be extract from https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/branches?base=hdf5-1_10_4, unless there is some better idea...

Thanks!


```
$ strings  libhdf5.103.dylib | grep 'HDF5 library version'
Warning! ***HDF5 library version mismatched error***
HDF5 library version: %d.%d.%d
The HDF5 library version information are not consistent in its source code.
HDF5 library version: 1.10.4
HDF5 library version: 1.10.4
```

Unfotunately, the HDF5 header files from Linux-ARM require the existence of a `feature.h` file which is not the case for `x86_64-w64-mingw32`. The `hdf5.h` file in the updated `HDF5_jll file` can thus not be used on Windows.

```
configure:17968: cc -c -I/workspace/destdir/include  conftest.c >&5
In file included from /workspace/destdir/include/hdf5.h:22:0,
                 from conftest.c:125:
/workspace/destdir/include/H5public.h:35:80: fatal error: features.h: No such file or directory
 #include <features.h>           /*for setting POSIX, BSD, etc. compatibility */
                                                                                ^
compilation terminated.
```

Here I compare the H5pubconf.h from ARM (currently installed) and from mingw64:

```
sandbox:${WORKSPACE}/srcdir/netcdf-c-4.7.2/mingw64/include # grep -i feature /workspace/srcdir/netcdf-c-4.7.2/mingw64/include/H5pubconf.h /workspace/destdir/include/H5pubconf.h 
/workspace/srcdir/netcdf-c-4.7.2/mingw64/include/H5pubconf.h:/* Define to 1 if you have the <features.h> header file. */
/workspace/srcdir/netcdf-c-4.7.2/mingw64/include/H5pubconf.h:/* #undef H5_HAVE_FEATURES_H */
/workspace/destdir/include/H5pubconf.h:/* Define to 1 if you have the <features.h> header file. */
/workspace/destdir/include/H5pubconf.h:#define H5_HAVE_FEATURES_H 1
```

Actually, the mingw64 package includes the header files. I would propose to use these on Windows.

--- a/H/HDF5/build_tarballs.jl
+++ b/H/HDF5/build_tarballs.jl
@@ -79,7 +79,6 @@ elif [[ ${target} == *apple* ]]; then
     done
 fi
 
-
 # We need to be able to access `libhdf5` and `libhdf5_hl` directly, so symlink it from the hashed filename from manylinux pypi
 if [[ ${target} == *86*linux* ]]; then
     libhdf5name=$(basename ${prefix}/lib/libhdf5-*.${dlext}*)
@@ -93,17 +92,19 @@ if [[ ${target} == *86*linux* ]]; then
     ln -s ${libhdf5_hlname} ${prefix}/lib/libhdf5_hl${ext}
 fi
 
-if [[ "${target}" == arm-linux-gnueabihf ]]; then
-    install_license ${prefix}/share/COPYING
-else
-    # Install headers using those from the ARM build
+if [[ "${target}" != arm-linux-gnueabihf ]]; then
+    # Install headers
     mkdir -p "${prefix}/include"
-    cp ${WORKSPACE}/srcdir/hdf5-arm-linux-gnueabihf-*/include/* "${prefix}/include"
-
-    # Remove the hash from license file name and then install it
-    mv ${WORKSPACE}/srcdir/*-COPYING ${WORKSPACE}/srcdir/COPYING
-    install_license ${WORKSPACE}/srcdir/COPYING
+    if [[ "${target}" == *-mingw* ]]; then
+        # Use MinGW header files, which of course are different
+        # from those for the other operating systems.
+        cp -r mingw${nbits}/include/* "${prefix}/include"
+    else
+        # Use headers from the ARM build, with the hope that they'll be fine
+        cp ${WORKSPACE}/srcdir/hdf5-arm-linux-gnueabihf-*/include/* "${prefix}/include"
+    fi
 fi
+install_license ${WORKSPACE}/srcdir/hdf5-arm-linux-gnueabihf-*/share/COPYING
 """
 
 # These are the platforms we will build for by default, unless further

